src/ms-store-badge.ts

```diff
@@ -242,8 +242,8 @@
       return MSStoreBadge.getSupportedLanguageFromUserAgent();
     }
 
-    // See if the language code is a supported language.
-    const supportedLanguage = MSStoreBadge.supportedLanguages.find(l => l.code.substring(0, 2) === languageCode.toLowerCase()) || MSStoreBadge.supportedLanguages.find(l => l.code === languageCode.toLowerCase());
+    // See if the language code is a supported language. First check for normal case (2 letter codes), then check for first edge case (same language letter codes, compare entire codes), then check for second edge case (3 letter codes; e.g. fil)
+    const supportedLanguage = MSStoreBadge.supportedLanguages.find(l => l.code.substring(0, 2) === languageCode.toLowerCase()) || MSStoreBadge.supportedLanguages.find(l => l.code === languageCode.toLowerCase()) || MSStoreBadge.supportedLanguages.find(l => l.code.substring(0, 3) === languageCode.toLowerCase());
```

Comment:
I think this won't work.

Consider the `fil` case: the first check will see if "fil".substring(0, 2) matches a language code...it does, but oops, it's Finnish (`fi`), not Filipino (`fil`).

In short, I think you need to reorder the checks here.